documentation/src/docs/getting-started/dev-guide.md

```diff
@@ -0,0 +1,31 @@
+# Getting Started Guide for Developers
+
+## Walk through first code sample
+
+## Walk through client and stream semantics
+
+### Ordered writes
+
+### Unordered writes
+
+### Batch client
+
+### Byte streams
+
+## Introducing Connectors
```

Comment:
This document will be huge if we also include connectors. Perhaps, we can create a separate document called `dev`-connectors-guide` or something to split it and make it more readable.